src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -322,6 +325,17 @@ internal void ToString(TraceFormat traceFormat, StringBuilder sb)
                             sb.Append(' ');
                             sb.AppendFormat(CultureInfo.InvariantCulture, inFileLineNum, fileName, sf.GetFileLineNumber());
                         }
+                        else if (LocalAppContextSwitches.ILOffsetToStackTrace && mb.ReflectedType != null)
+                        {
+                            string assemblyName = System.IO.Path.GetFileName(mb.ReflectedType.Assembly.Location);
```

Comment:
```suggestion
                            string assemblyName = mb.ReflectedType.Module.ScopeName;
```

`Assembly.Location` is going to be an empty string in [single file applications](https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file#api-incompatibility). ScopeName is the name from metadata. It will be the same as the file name most of the time, unless someone renamed the file after compilation.